cmd/proxy-to-grafana/proxy-to-grafana.go

```diff
@@ -134,27 +203,43 @@ func modifyRequest(req *http.Request, localClient *local.Client) {
 		return
 	}
 
-	user, err := getTailscaleUser(req.Context(), localClient, req.RemoteAddr)
+	user, role, err := getTailscaleIdentity(req.Context(), localClient, req.RemoteAddr)
 	if err != nil {
 		log.Printf("error getting Tailscale user: %v", err)
 		return
 	}
 
 	req.Header.Set("X-Webauth-User", user.LoginName)
 	req.Header.Set("X-Webauth-Name", user.DisplayName)
+	if role != UnknownRole {
+		req.Header.Set("X-Webauth-Role", role.String())
```

Comment:
do we strip this header from the incoming request anywhere to prevent role spoofing?

Reply:
No, we don't. I've adjusted the PR to default to `ViewerRole` and always set this header. WDYT?

Reply:
That works. But what happens if `getTailscaleIdentity` fails, like getting a request from a tagged node? Would it be able to inject any `X-Webauth-*` headers because we don't abort the request?
Maybe we should either drop the request on error, or at least preemptively strip all `X-Webauth-*` headers in this function?

Reply:
Oh right, I thought we were failing the request on error, but now I can see that `return` statement just 6 lines earlier 😅 

I've added explicit deletion of `X-Webauth-*` headers.